src/lib/flight_tasks/tasks/AutoLineSmoothVel/FlightTaskAutoLineSmoothVel.cpp

```diff
@@ -328,8 +329,20 @@ void FlightTaskAutoLineSmoothVel::_updateTrajConstraints()
 	_trajectory[2].setMaxJerk(_param_mpc_jerk_auto.get());
 
 	if (_velocity_setpoint(2) < 0.f) { // up
-		_trajectory[2].setMaxAccel(_param_mpc_acc_up_max.get());
-		_trajectory[2].setMaxVel(_param_mpc_z_vel_max_up.get());
+		float z_accel_constraint = _param_mpc_acc_up_max.get();
+		float z_vel_constraint = _param_mpc_z_vel_max_up.get();
+
+		// The constraints are broken because they are used as hard limits by the position controller, so put this here
+		// until the constraints don't do things like cause controller integrators to saturate. Once the controller
+		// doesn't use z speed constraints, this can go in AutoMapper::_prepareTakeoffSetpoints(). Accel limit is to
+		// emulate the motor ramp (also done in the controller) so that the controller can actually track the setpoint.
+		if (_type == WaypointType::takeoff &&  _dist_to_ground < _param_mpc_land_alt1.get()) {
```

Comment:
I think this should be `MPC_LAND_ALT2` , the closest distance to the ground. Alt1 if higher up.

Reply:
Makes sense